drools-wb-screens/drools-wb-enum-editor/drools-wb-enum-editor-client/src/test/java/org/drools/workbench/screens/enums/client/editor/EnumEditorPresenterTests.java

```diff
@@ -0,0 +1,217 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.enums.client.editor;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.enterprise.event.Event;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwtmockito.GwtMock;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.drools.workbench.screens.enums.client.type.EnumResourceType;
+import org.drools.workbench.screens.enums.model.EnumModel;
+import org.drools.workbench.screens.enums.model.EnumModelContent;
+import org.drools.workbench.screens.enums.service.EnumService;
+import org.guvnor.common.services.shared.metadata.model.Metadata;
+import org.guvnor.common.services.shared.metadata.model.Overview;
+import org.jboss.errai.common.client.api.Caller;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.widgets.client.menu.FileMenuBuilder;
+import org.kie.workbench.common.widgets.client.source.ViewDRLSourceWidget;
+import org.kie.workbench.common.widgets.metadata.client.KieEditorWrapperView;
+import org.kie.workbench.common.widgets.metadata.client.widget.OverviewWidgetPresenter;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.uberfire.backend.vfs.ObservablePath;
+import org.uberfire.ext.editor.commons.client.history.VersionRecordManager;
+import org.uberfire.ext.editor.commons.client.validation.DefaultFileNameValidator;
+import org.uberfire.mocks.CallerMock;
+import org.uberfire.mocks.EventSourceMock;
+import org.uberfire.mvp.Command;
+import org.uberfire.mvp.PlaceRequest;
+import org.uberfire.workbench.events.NotificationEvent;
+import org.uberfire.workbench.model.menu.MenuItem;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class EnumEditorPresenterTests {
+
+    @Mock
+    private EnumEditorView view;
+
+    @Mock
+    private KieEditorWrapperView mockKieView;
+
+    @Mock
+    private OverviewWidgetPresenter mockOverviewWidget;
+
+    @Mock
+    private VersionRecordManager mockVersionRecordManager;
+
+    @Mock
+    private FileMenuBuilder mockFileMenuBuilder;
+
+    @Mock
+    private DefaultFileNameValidator mockFileNameValidator;
+
+    @Mock
+    private EnumService enumService;
+    private Caller<EnumService> enumServiceCaller;
+
+    @Mock
+    private ObservablePath path;
+
+    @Mock
+    private PlaceRequest place;
+
+    @Mock
+    private Overview overview;
+
+    @GwtMock
+    private ViewDRLSourceWidget sourceWidget;
+
+    @Captor
```

Comment:
Maybe I should read the Mockito docs again. Didn't realize I could do this. This is a lot nicer way.